internal/scrapertest/mask.go

```diff
@@ -0,0 +1,51 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scrapertest // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/scrapertest"
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+// MaskMetricSliceValues sets all data point values to zero.
+func MaskMetricSliceValues(metrics pdata.MetricSlice) {
+	for i := 0; i < metrics.Len(); i++ {
+		MaskMetricValues(metrics.At(i))
+	}
+}
+
+// MaskMetricValues sets all data point values to zero.
+func MaskMetricValues(metric pdata.Metric) {
+	var dataPoints pdata.NumberDataPointSlice
+	switch metric.DataType() {
+	case pdata.MetricDataTypeGauge:
+		dataPoints = metric.Gauge().DataPoints()
+	case pdata.MetricDataTypeSum:
+		dataPoints = metric.Sum().DataPoints()
+	default:
+		panic(fmt.Sprintf("data type not supported: %s", metric.DataType()))
+	}
+	MaskDataPointSliceValues(dataPoints)
+}
+
+// MaskDataPointSliceValues sets all data point values to zero.
+func MaskDataPointSliceValues(dataPoints pdata.NumberDataPointSlice) {
```

Comment:
The one issue I'm seeing with this is, currently we are still checking to see if the type (double, int) is correct, can you think of a way to be able to preserve that to ensure the right datatype is being used?

Reply:
It just sets both values to 0, so comparisons ignore it regardless. 

I don't think the data model is intended to distinguish between the two really, so I don't think this is something we necessarily need to worry about. In any case, it's not a regression from what was previously here.